From 1c1a02632f0d3a779f7f31729c7fa85ccedf0415 Mon Sep 17 00:00:00 2001
From: "Patil, Rachna" <rachna@ti.com>
Date: Wed, 28 Sep 2011 16:31:50 +0530
Subject: [PATCH 0557/1092] AM335X: Avoid i2c pin mux setup.

Since i2c1 is available only in some profiles, we bypass i2c pin
muxing in i2c.c file. Pin muxing is taken care of, in the board file
instead.

Signed-off-by: Patil, Rachna <rachna@ti.com>
Signed-off-by: Vaibhav Hiremath <hvaibhav@ti.com>
---
 arch/arm/plat-omap/i2c.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/arch/arm/plat-omap/i2c.c b/arch/arm/plat-omap/i2c.c
index 6cddde2..b7fccc8 100644
--- a/arch/arm/plat-omap/i2c.c
+++ b/arch/arm/plat-omap/i2c.c
@@ -148,7 +148,8 @@ static inline int omap2_i2c_add_bus(int bus_id)
 	struct omap_i2c_bus_platform_data *pdata;
 	struct omap_i2c_dev_attr *dev_attr;
 
-	omap2_i2c_mux_pins(bus_id);
+	if (!cpu_is_am33xx())
+		omap2_i2c_mux_pins(bus_id);
 
 	l = snprintf(oh_name, MAX_OMAP_I2C_HWMOD_NAME_LEN, "i2c%d", bus_id);
 	WARN(l >= MAX_OMAP_I2C_HWMOD_NAME_LEN,
-- 
1.7.11.2

